pkg/services/ngalert/notifier/external_alertmanager.go

```diff
@@ -0,0 +1,278 @@
+package notifier
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	alertingNotify "github.com/grafana/alerting/notify"
+	"github.com/grafana/grafana/pkg/infra/log"
+	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"gopkg.in/yaml.v3"
+)
+
+type externalAlertmanager struct {
+	url      string
+	username string
+	password string
+}
+type configStore interface {
+	GetLatestAlertmanagerConfiguration(ctx context.Context, query *models.GetLatestAlertmanagerConfigurationQuery) (*models.AlertConfiguration, error)
+}
+
+// TODO: naming...
+type externalMultiOrgAlertmanager struct {
+	log log.Logger
+	// Note: this is not multi-org! We're just creating one for the main org.
+	alertmanager *externalAlertmanager
+	store        configStore
+}
+
+type alwkdnaw interface {
+}
+
+func NewExternalMultiOrgAlertmanager(url, username, password string, store configStore, log log.Logger) *externalMultiOrgAlertmanager {
+	am := externalAlertmanager{
+		url:      url,
+		username: username,
+		password: password,
+	}
+
+	return &externalMultiOrgAlertmanager{
+		log:          log,
+		store:        store,
+		alertmanager: &am,
+	}
+}
+
+// TODO: we're completely ignoring orgID.
+func (am *externalMultiOrgAlertmanager) AlertmanagerFor(orgID int64) (Alertmanager, error) {
+	// TODO: main OrgID always 1?
+	fmt.Println("noop.AlertmanagerFor() called")
+	if orgID == 1 {
```

Comment:
tenantID, ok := tenant_mappers[orgID]
if !ok {
return errors.New("there is not tenant set for that org, please correct tenant_mappings")
}

request.TenantID = tenantID

All requests made by this are set to `X-Scope-OrgID: tenantID`